lib/libc/wasi/libc-bottom-half/sources/preopens.c

```diff
@@ -100,35 +101,42 @@ static const char *strip_prefixes(const char *path) {
     return path;
 }
 
-/// Register the given preopened file descriptor under the given path.
-///
-/// This function takes ownership of `prefix`.
-static int internal_register_preopened_fd(__wasi_fd_t fd, const char *relprefix) {
-    LOCK(lock);
-
+/// Similar to `internal_register_preopened_fd_unlocked` but does not
```

Comment:
I think this comment is meant to cite,
```suggestion
/// Similar to `internal_register_preopened_fd` but does not
```

Reply:
This is part of upstream `wasi-libc`, so it should be fixed there :)